core/tasks/model_monitoring.py

```diff
@@ -0,0 +1,160 @@
+        description=model_info["description"],
+        reference_train_data=model_info["reference_train_data"],
+    )
+
+    model_instances = [
+        ClassificationPredictions(
+            entry=Entry.objects.get(original_entry_id=int(record["entry_id"])),
+            project=Project.objects.get(
+                lead=Lead.objects.get(
+                    entry=Entry.objects.get(original_entry_id=int(record["entry_id"]))
+                )
+            ),
+            model=model,
+            embeddings=record["embeddings"],
+            subpillars_1d=record["subpillars_1d_pred"],
+            sectors=record["sectors_pred"],
+            subpillars_2d=record["subpillars_2d_pred"],
+            age=record["age_pred"],
+            gender=record["gender_pred"],
+            affected_groups=record["affected_groups_pred"],
+            specific_needs_groups=record["specific_needs_groups_pred"],
+            severity=record["severity_pred"],
+        )
+        for record in df
+    ]
+
+    predictions = ClassificationPredictions.objects.bulk_create(model_instances)
+    return predictions
+
+
+def save_dataframe_to_model(dataframe, model_class):
+    """
+    Saves a pandas DataFrame into a Django model.
+    Args:
+        dataframe (pandas.DataFrame): The DataFrame to save.
+        model_class (django.db.models.Model): The Django model class to save the DataFrame to.
+    """
+    for row in dataframe.itertuples(index=False):
+        model_instance = model_class()
+        for field in dataframe.columns:
+            setattr(model_instance, field, getattr(row, field))
+        model_instance.save()
+
+
+def create_model_performance(df):
+    modelperf = ModelPerformance(df)
+    df1 = modelperf.project_wise_perf_metrics()
+    save_dataframe_to_model(df1, ProjectWisePerfMatrices)
+
+    df2 = modelperf.per_tag_perf_metrics()
+    save_dataframe_to_model(df2, TagWisePerfMatrics)
+
+    df3 = modelperf.all_projects_perf_metrics()
+    save_dataframe_to_model(df3, AllProjectPerfMatrics)
+
+    df4 = modelperf.calculate_ratios()
+    save_dataframe_to_model(df4, CategoryWiseMatchRatios)
+
+    df5 = modelperf.per_project_calc_ratios()
+    save_dataframe_to_model(df5, ProjectWiseMatchRatios)
+
+
+def set_project_id(row):
+    project = Project.objects.get(
+        lead=Lead.objects.get(
+            entry=Entry.objects.get(original_entry_id=row["entry_id"])
+        )
+    )
+    return project.original_project_id
+
+
+@shared_task
+def calculate_model_metrics():
+    with transaction.atomic():
+        # create a data frame
+        df = pd.DataFrame(
+            list(
+                Entry.objects.exclude(
+                    original_entry_id__in=ClassificationPredictions.objects.values_list(
+                        "entry__original_entry_id", flat=True
+                    )
+                )
+                .order_by("-id")
+                .values("original_entry_id", "excerpt_en", "original_af_tags")
+            )[0:10]
```

Comment:
Instead of slicing on list, please do that on the queryset itself. That's much efficient.